sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -628,12 +628,17 @@ def _cleanup(self, on_exit=False):
           '  with Pipeline() as p:\n'
           '    p.apply(..)\n'
           'This ensures that the pipeline finishes before this program exits.')
-    has_exception = None
+    callback_exceptions = []
     for callback in self._cleanup_callbacks:
       try:
         callback()
-      except Exception:
-        has_exception = True
+      except Exception as e:
+        callback_exceptions.append(e)
+
     self._cleanup_callbacks = ()
-    if has_exception:
-      raise
+    if callback_exceptions:
+      formatted_exceptions = ''.join(
+          [f"\n\t{repr(e)}" for e in callback_exceptions])
+      raise RuntimeError(
+          'Cleanup callbacks raised the following errors: {}'.format(
```

Comment:
May be we just mention errors instead of `Cleanup callbacks` as  `_cleanup` is an internal method and may be it creates confusion among users who tries to look at the logs.